mzLib/TestFlashLFQ/TestFlashLFQ.cs

```diff
@@ -916,26 +918,153 @@ public static void TestBayesianProteinQuantification()
+            SpectraFileInfo f1r2 = new SpectraFileInfo(Path.Combine(testDataDirectory, "20100614_Velos1_TaGe_SA_K562_4.mzML"), "two", 1, 1, 1);
+
+            List<string> acceptableProteinGroupAccessions = new() { "Q7KZF4", "Q15149", "P52298" };
+
+            List<Identification> ids = new List<Identification>();
+            Dictionary<string, ProteinGroup> allProteinGroups = new Dictionary<string, ProteinGroup>();
+            foreach (string line in File.ReadAllLines(psmFile))
+            {
+                var split = line.Split(new char[] { '\t' });
+
+                //skip the header
+                if (split.Contains("File Name") || string.IsNullOrWhiteSpace(line))
+                {
+                    continue;
+                }
+
+                SpectraFileInfo file = null;
+
+                if (split[0].Contains("20100614_Velos1_TaGe_SA_K562_3"))
+                {
+                    file = f1r1;
+                }
+                else if (split[0].Contains("20100614_Velos1_TaGe_SA_K562_4"))
+                {
+                    file = f1r2;
+                }
 
-            //// try with paired samples
-            //peptide.SetIntensity(files[0], 100);
-            //peptide.SetIntensity(files[1], 1000);
-            //peptide.SetIntensity(files[2], 10000);
+                string baseSequence = split[12];
+                string fullSequence = split[13];
+                double monoMass = double.Parse(split[22]);
+                double rt = double.Parse(split[2]);
+                int z = (int)double.Parse(split[6]);
+                var proteinSubset = split[25].Split(new char[] { '|' });
+                List<ProteinGroup> proteinGroups = new();
 
-            //peptide.SetIntensity(files[3], 210);
-            //peptide.SetIntensity(files[4], 2200);
-            //peptide.SetIntensity(files[5], 21500);
+                if (acceptableProteinGroupAccessions.Contains(proteinSubset.First()))
+                {
+                    foreach (var protein in proteinSubset)
+                    {
+                        if (allProteinGroups.TryGetValue(protein, out var proteinGroup))
+                        {
+                            proteinGroups.Add(proteinGroup);
+                        }
+                        else
+                        {
+                            allProteinGroups.Add(protein, new ProteinGroup(protein, "", ""));
+                            proteinGroups.Add(allProteinGroups[protein]);
+                        }
+                    }
+
+                    Identification id = new Identification(file, baseSequence, fullSequence, monoMass, rt, z, proteinGroups);
+                    ids.Add(id);
+                }
+            }
+
+            var engine = new FlashLfqEngine(ids, matchBetweenRuns: true, requireMsmsIdInCondition: false, useSharedPeptidesForProteinQuant: true, maxThreads: 1);
```

Comment:
Could squeeze more lines of tested code out if you set maxthreads to -1 here. Should reset it to 1 in the code.

Reply:
changed